ui/shared/components/panel/variants/selectors.js

```diff
@@ -130,7 +131,15 @@ export const getPairedSelectedSavedVariants = createSelector(
       return acc
     }, [])
 
-    if (tag === NOTE_TAG_NAME) {
+    const tags = (tag || '').split(TAG_URL_DELIMITER)
+    if (tags.length > 1) {
+      pairedVariants = matchingVariants(
+        pairedVariants, ({ tagGuids }) => {
+          const tagNames = tagGuids.map(tagGuid => tagsByGuid[tagGuid].name)
+          return tags.every(tagName => tagNames.includes(tagName))
```

Comment:
In the end, we still need to add filtering for multiple tags.

Reply:
Why? The combined DB filter you had implemented in the last round of review worked?

Reply:
We have the state for the tag loaded but don't organize the variants per tag. This means all loaded variants are still available when the tag is changed. So we must filter the variants to display according to the current tag.

Reply:
so don't set the "tag loaded" state when multiple tags are queried, and always refetch for multiple tags. We are going to run into untenable performance problems if we use AND instead of OR in the database query, so adding a tiny frontend optimization to prevent a couple extra backend requests is not worth the tradeoff of having actual needed queries time out because they return twice as much data instead of half

Reply:
We don't refetch for every tag change because we have the tag-loaded state. But the loaded variants won't be removed when more variants are loaded. We need some code to filter out the variants to display in the front end. We could add/update the reducer to store the variants per tag(s). But we can also use the selectors to organize the variants. That's pretty much the current code.

Reply:
for multiple tags, always load variants and then use the results of that load as the variants to show. Do not use any of the other variants that have been previously loaded. Put the results of a multi-tag search into a different field in the state and then use that if its present and don;t use the other state fields for the variant list

Reply:
This should bevery similar to how `searchedVariants` are handled for variant searches

Reply:
Then, we may be unable to use the `SavedVariants` shared component.

Reply:
They already use different code to load saved variants, which means the code for where in the state to put the results after loading is already separate. Changing the selector to look first and see if theres data in the "use this if populated" part of the state, and then falling back to the current logic if not would be very straightforward to add

Reply:
theres also already a `projectGuid` prop in the selector for what variants to show, so you could also make it conditional on that